virttest/utils_test/libvirt.py

```diff
@@ -1056,8 +1057,14 @@ def pre_pool(self, pool_name, pool_type, pool_target, emulated_image,
                     iscsi_target = iscsi_node[1]
                     break
             iscsi.iscsi_logout(iscsi_target)
-            extra += " --source-host %s  --source-dev %s" % (ip_addr,
-                                                             iscsi_target)
+            if pool_type == "iscsi-direct":
+                extra += " --source-host %s --source-dev %s " \
+                         "--source-iqn %s" % (ip_addr, iscsi_target,
+                                              iscsi_initiator)
+            else:
+                extra += " --source-host %s  --source-dev %s" % (ip_addr,
+                                                                 iscsi_target)
```

Comment:
I think only add --source-iqn for iscsi-direct pool is fine here.
            if pool_type == "iscsi-direct":
                extra += "--source-iqn %s" %  iscsi_initiator

Reply:
Updated, thaks for your useful suggestions.